packages/back-end/src/controllers/savedGroups.ts

```diff
@@ -0,0 +1,75 @@
+import { AuthRequest } from "../types/AuthRequest";
+import { Response } from "express";
+import { getOrgFromReq } from "../services/organizations";
+import {
+  createSavedGroup,
+  parseSaveGroupString,
+  updateSavedGroup,
+} from "../models/SavedGroupModel";
+
+// IMPORTANT: SavedGroups and Groups are very similar, but serve two different purposes. At the time of development 9/22 we are
+// quietly deprecating Groups. Initially groups were used with experiments to only include people in a group in an experiement.
+// SavedGroups are used with features flag rules where rules can say if "x is/is not in SavedGroup" do/don't show a feature
+
+export async function postSavedGroup(
+  req: AuthRequest<{
+    groupName: string;
+    owner: string;
+    attributeKey: string;
+    groupList: string;
+  }>,
+  res: Response
+) {
+  const { org } = getOrgFromReq(req);
+  const { groupName, owner, attributeKey, groupList } = req.body;
+
+  req.checkPermissions("createFeatures");
+
+  const values = parseSaveGroupString(groupList);
+
+  const savedGroup = await createSavedGroup({
+    values,
+    groupName,
+    owner,
+    attributeKey,
+    organization: org.id,
+  });
+
+  return res.status(200).json({
+    status: 200,
+    savedGroup,
+  });
+}
+
+export async function putSavedGroup(
+  req: AuthRequest<{
+    id: string;
+    groupName: string;
+    owner: string;
+    attributeKey: string;
+    groupList: string;
+  }>,
+  res: Response
+) {
+  const { org } = getOrgFromReq(req);
+  const { id, groupName, owner, groupList, attributeKey } = req.body;
+
+  if (!id) {
+    throw new Error("Must specify saved group id");
+  }
+
+  req.checkPermissions("createFeatures");
+
+  const values = parseSaveGroupString(groupList);
+
+  await updateSavedGroup(id, org.id, {
+    values,
+    groupName,
+    owner,
+    attributeKey,
```

Comment:
This still seems risky to allow changing the attributeKey.  If the saved group is being used in a feature already, changing the attributeKey will break things.  And it's really hard to detect if it's being used - we can't just do a simple Mongo query since the group ids are embedded inside of strings.

If someone made a mistake and really does want to change the attribute key, they can just delete the saved group and create a new one.  It's a little annoying, but I suspect it's super rare and the alternative seems worse - accidentally messing up existing features in unpredictable ways.

Reply:
I went back and forth on this - I can definitely see the use case you entered being a really tough bug to potentially catch. My initial thought was if someone did change it, when we went to evaluate the feature rules, it would just result in the rule not being evaluated as true, sorta like if the list had been deleted.

But, I could see some weird states where weird things might happen. I'll remove the attributeKey as an editable property.

Reply:
Updated to make it so attributeKey is not editable.